src/ray/gcs/gcs_server/gcs_node_manager.cc

```diff
@@ -83,10 +83,10 @@ void GcsNodeManager::HandleGetAllNodeInfo(const rpc::GetAllNodeInfoRequest &requ
                                           rpc::GetAllNodeInfoReply *reply,
                                           rpc::SendReplyCallback send_reply_callback) {
   for (const auto &entry : alive_nodes_) {
-    reply->add_node_info_list()->CopyFrom(*entry.second);
+    reply->mutable_node_info_list()->UnsafeArenaAddAllocated(entry.second.get());
```

Comment:
Can you add a comment as to why these unsafe adds are OK? (presumably since both messages were allocated in the same arena?)

Reply:
Actually, not because of that. It's because entry's life cycle is longer than reply. I'll comment on it there.